k8s/pkg/apis/longhorn/v1beta1/backingimage.go

```diff
@@ -18,33 +18,48 @@ const (
 )
 
 type BackingImageDiskFileStatus struct {
-	State                   BackingImageState `json:"state"`
-	Progress                int               `json:"progress"`
-	Message                 string            `json:"message"`
-	LastStateTransitionTime string            `json:"lastStateTransitionTime"`
+	// +optional
+	State BackingImageState `json:"state"`
+	// +optional
+	Progress int `json:"progress"`
+	// +optional
+	Message string `json:"message"`
+	// +optional
+	LastStateTransitionTime string `json:"lastStateTransitionTime"`
 }
 
 type BackingImageSpec struct {
-	Disks            map[string]string          `json:"disks"`
-	Checksum         string                     `json:"checksum"`
-	SourceType       BackingImageDataSourceType `json:"sourceType"`
-	SourceParameters map[string]string          `json:"sourceParameters"`
-
+	// +optional
+	Disks map[string]string `json:"disks"`
+	// +optional
+	Checksum string `json:"checksum"`
+	// +optional
```

Comment:
It's better to mark some fields are required if possible. We can do it later with a separate ticket tracking it.

Reply:
Yes, that would be great. But since lots of spec fields need to check one by one which is/are required, the only way I could think now is run the E2E test but it required lots of time. 

I agree that we should enhance it in the future.